### PR TITLE
Activate linting with black

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

Reduce neatpicking during code reviews and stop worrying about code formatting.

## Proposal

Adopt Black as our formatting standard. The only customization is increasing max line length to 99.

Developpers are expected to run Black on their laptop before committing their code.

The compliance of code formatting with Black's rules is enforced during the CI build.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -334,7 +335,7 @@ ignore-docstrings=yes
 ignore-imports=yes
 
 # Minimum lines number of a similarity.
-min-similarity-lines=6
+min-similarity-lines=7
```
[the remark this answers is not in the document]
Comment (sampaccoud) [reply .pylintrc]: We have been increasing it slowly... I think it is still valuable to catch a real copy/paste so it doesn't hurt to keep it... ?

**Comment by sampaccoud** [comment]
I would like to open an issue to rewrite the README in a separate PR as it is currently outdated and there is nowhere to talk about Black without a major rewrite.

File: .pylintrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmaupetit) [reply .pylintrc]: No it doesn't, you are right.

**Comment by jmaupetit** [comment]
True. Let's make Docker instructions the default documentation / README. "container-native development" they say :wink:

**Pull request merged**